### PR TITLE
handlers: Missing resources should return 404

### PR DESCRIPTION
**Pull request by samuelkarp**
- [x] Build
- [x] Unit tests
- [x] Integ tests
- [x] Functional tests

r? @aaithal @juanrhenals @richardpen

### CODE REVIEW

**Comment by samuelkarp** [comment]
Integ tests appear to be failing in an unrelated way.  I'll open a separate issue for that.

File: agent/handlers/v1_handlers_test.go [hunk 1]
```diff
@@ -78,11 +78,11 @@ func TestGetTaskByDockerID(t *testing.T) {
 	taskDiffHelper(t, []*api.Task{testTasks[1]}, TasksResponse{Tasks: []*TaskResponse{&taskResponse}})
 }
 
-func TestGetTaskByDockerID400(t *testing.T) {
+func TestGetTaskByDockerID404(t *testing.T) {
```
Comment (aaithal) [inline agent/handlers/v1_handlers_test.go]: Can we rename this to not contain the response code in the test name?

File: agent/handlers/v1_handlers_test.go [hunk 2]
```diff
@@ -78,11 +78,11 @@ func TestGetTaskByDockerID(t *testing.T) {
...
 	recorder := performMockRequest(t, "/v1/tasks?dockerid=does-not-exist")
 
-	if recorder.Code != 400 {
-		t.Error("API did not return 400 for bad dockerid")
+	if recorder.Code != 404 {
```
Comment (aaithal) [inline agent/handlers/v1_handlers_test.go]: Also, we can use `http.StatusBadRequest` instead of the "magic" number

**Comment by aaithal** [comment]
lgtm, some minor comments.

**Comment by richardpen** [comment]
👍

**Pull request merged**